### PR TITLE
Jax implementation of `log1mexp`

### LINKED ISSUE #990: Add JAX implementation for log1mexp
It seems like there is no JAX implementation for the `log1mexp` Op. Should be easy to add.

### PR DESCRIPTION
**Pull request by kylejcaron**
This PR creates a Jjax implementation of the log1mexp op, to resolve issue #990 

**Thank you for opening a PR!**

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [ ] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [x] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [x] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

**Comment by kylejcaron** [comment]
quick callout: `pre-commit` had the following failure unrelated to any code introduced in my PR so I uninstalled `pre-commit` and proceeded to commit my work

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

aesara/scan/op.py:3431: error: Unused "type: ignore" comment
Found 1 error in 1 file (checked 2 source files)
```

File: tests/link/test_jax.py [hunk 1]
```diff
@@ -1371,3 +1371,11 @@ def test_psi():
     out = psi(x)
     fg = FunctionGraph([x], [out])
     compare_jax_and_py(fg, [3.0])
+
+
+def test_log1mexp():
+    x = scalar("x")
```
Comment (ricardoV94) [inline tests/link/test_jax.py:1377]: Can you use a vector and test values on the two branches of the `where` statement?

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Looks good, I have a suggestion below for a more thorough check

**Comment by ricardoV94** [comment]
> quick callout: `pre-commit` had the following failure unrelated to any code introduced in my PR so I uninstalled `pre-commit` and proceeded to commit my work
> 
> ```
> mypy.....................................................................Failed
> - hook id: mypy
> - exit code: 1
> 
> aesara/scan/op.py:3431: error: Unused "type: ignore" comment
> Found 1 error in 1 file (checked 2 source files)
> ```

I have seen that can sometimes happen when the pre-commit runs only on the modified files but goes away when it runs on all files. It's annoying though

File: tests/link/test_jax.py (hunk 1 above)
Comment (kylejcaron) [reply tests/link/test_jax.py]: updated! Now test 2 values on each branch of the `where` statement

**Comment by ricardoV94** [comment]
Looks good, can you squash the three commits into a single one?

**Comment by kylejcaron** [comment]
> Looks good, can you squash the three commits into a single one?

just squashed into 1 commit!

**Review by ricardoV94 — DISMISSED** [review]

**Comment by kylejcaron** [comment]
> Thanks @kylejcaron!

Thanks for the help on this!

**Pull request merged**